### PR TITLE
Add Itu SP Spider

### PR DESCRIPTION
**Pull request by gabubellon**
Hello !!
Add Itu -SP spider.

A detail about this spider, Itu official [site](https://itu.sp.gov.br/imprensa-oficial/) use a javascript API from [DOSP](https://dosp.com.br) that return a JSON with journal details (ids and link), and use another API to parser a call to journal pdf.

Therefore I use a `chompjs` lib to parse javascript to a dict and parse data based on oficial [documentation](https://docs.scrapy.org/en/latest/topics/dynamic-content.html#parsing-javascript-code).

### CODE REVIEW

**Comment by adorilson** [comment]
This PR is part of #298

File: processing/data_collection/gazette/spiders/sp_itu.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import dateparser
+import chompjs
+import base64
+
+from datetime import datetime
+from scrapy import Request, Spider
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
```
Comment (giuliocc) [inline processing/data_collection/gazette/spiders/sp_itu.py:8]:
```suggestion
import base64
from datetime import datetime

import chompjs
import dateparser
from scrapy import Request, Spider

from gazette.items import Gazette
from gazette.spiders.base import BaseGazetteSpider
```

Just to follow PEP8's import rule.

File: processing/requirements.txt [hunk 2]
```diff
@@ -54,6 +54,7 @@ tzlocal==2.1              # via dateparser
 urllib3==1.25.10          # via botocore, requests
 w3lib==1.22.0             # via itemloaders, parsel, scrapy
 zope.interface==5.1.0     # via scrapy, twisted
+chompjs==1.0.16
```
Comment (giuliocc) [inline processing/requirements.txt:57]: Could you install `pip-tools` in your environment and then:
1. add `chompjs` to `requirements.in`
2. run `pip-compile requirements.in`
3. commit the modified `requirements.in` and `requirements.txt`

That way we can keep a more readable dependencies file (`requirements.in`) and a fixed dependency tree for an easy reproducible build (`requirements.txt`)

File: processing/data_collection/gazette/spiders/sp_itu.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+        response_js = chompjs.parse_js_object(response.text)
+
+        for element in response_js.get("data"):
+            print("DATE")
+            date = self.extract_date(element)
+
+            print("URL")
+            url = self.extract_url(element)
+            print(url)
```
Comment (giuliocc) [inline processing/data_collection/gazette/spiders/sp_itu.py:33]: I think the `print`s can be removed now :) They served their purpose

File: processing/data_collection/gazette/spiders/sp_itu.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+            yield Gazette(
+                date=date,
+                file_urls=[url],
+                territory_id=self.TERRITORY_ID,
+                power="executive_legislature",
+                scraped_at=datetime.utcnow(),
+            )
```
Comment (giuliocc) [inline processing/data_collection/gazette/spiders/sp_itu.py:41]: I think we can get the `edition_number` field from `"edicao_do"` too. Don't you think?

**Review by giuliocc — CHANGES_REQUESTED** [review]
Very nice and clean implementation :) That was great!

Just tested and everything is working as expected.

Just made some minor corrections there and suggestions for improvement.

As both you and @adorilson pointed out, this spider is part of a system that can be replicated for multiple spiders. After this PR I think we could make another PR transforming most of this into a base spider and adding spiders for each remaining city. Would you want to do that?

**Comment by gabubellon** [comment]
CR and remove autogenerated gazette fields

**Comment by gabubellon** [comment]
> Very nice and clean implementation :) That was great!
> 
> Just tested and everything is working as expected.
> 
> Just made some minor corrections there and suggestions for improvement.
> 
> As both you and @adorilson pointed out, this spider is part of a system that can be replicated for multiple spiders. After this PR I think we could make another PR transforming most of this into a base spider and adding spiders for each remaining city. Would you want to do that?

@giuliocc Thanks !
And Yes, I believe the next step is move to a base spider, but two things need check first:
* Get all the cities codes from DiOE (#298);
* Check if returned json are equals from all cities.
I like on work that !

**Review by giuliocc — APPROVED** [review]
LGTM! That was a great first contribution :)

If anyone else merges, remember to "Squash and Merge"

**Comment by giuliocc** [comment]
Thank you so much for the contribution!!

**Pull request merged**